robot_3d/src/joint_states_publisher_3d.cpp

```diff
@@ -0,0 +1,55 @@
+#include <string>
+#include <cstdlib>
+#include <ctime>
+#include <ros/ros.h>
+#include <sensor_msgs/JointState.h>
+#include "std_msgs/String.h"
+
+int main(int argc, char** argv)
+{
+  ros::init(argc, argv, "joint_states_publisher_3d");
+  ros::NodeHandle n;
+  ros::Publisher joint_pub = n.advertise<sensor_msgs::JointState>("joint_states", 1000);
+
+  // Gets the loop rate set in the launch file and sends it to ros::rate to set the rate at which the joint angles
+  // change.
+  double loop_rate;
+  n.getParam("loop_rate", loop_rate);
+  ros::Rate my_loop_rate(loop_rate);
+
+  sensor_msgs::JointState joint_state;
+  srand(time(0));
+
+  while (ros::ok)
+  {
+    joint_state.header.stamp = ros::Time::now();
+
+    // Randomly generating joint angles within ranges specified in .xacro file.
+    // Generating joint angles for joints 1 and 3 within the range [-3.1415, 3.1415].
+    double joint_1_3 = (rand() % 62831 + (-31415)) / 10000.0;
```

Comment:
Looks like the way you are randomly generating, joint pairs (1, 3) and (4, 6) will always have the same values, which removes some of the randomness of the simulation. 

Even though it may seem repetitive, it is actually the better design to generate all joint angles separately.

Also we should make a function `genRandomAngle(float low, float hi)` that follows this (more common formula for random number generation within float ranges: https://stackoverflow.com/a/686373